src/strands/multiagent/swarm.py

```diff
@@ -0,0 +1,646 @@
+- Autonomous agent collaboration without central control
+- Dynamic task distribution based on agent capabilities
+- Collective intelligence through shared context
+"""
+
+import asyncio
+import logging
+import time
+from concurrent.futures import ThreadPoolExecutor
+from dataclasses import dataclass, field
+from typing import Any, Callable, Tuple, cast
+
+from ..agent import Agent, AgentResult
+from ..tools.decorator import tool
+from ..types.event_loop import Metrics, Usage
+from .base import MultiAgentBase, MultiAgentResult, NodeResult, Status
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class SwarmNode:
+    """Represents a node (e.g. Agent) in the swarm."""
+
+    node_id: str
+    executor: Agent
+
+    def __hash__(self) -> int:
+        """Return hash for SwarmNode based on node_id."""
+        return hash(self.node_id)
+
+    def __eq__(self, other: Any) -> bool:
+        """Return equality for SwarmNode based on node_id."""
+        if not isinstance(other, SwarmNode):
+            return False
+        return self.node_id == other.node_id
+
+    def __str__(self) -> str:
+        """Return string representation of SwarmNode."""
+        return self.node_id
+
+    def __repr__(self) -> str:
+        """Return detailed representation of SwarmNode."""
+        return f"SwarmNode(node_id='{self.node_id}')"
+
+
+@dataclass
+class SwarmMessage:
+    """Message passed between nodes in swarm."""
+
+    from_node: SwarmNode
+    to_node: SwarmNode
+    content: str
+    context: dict[str, Any]
+    timestamp: float = field(default_factory=time.time)
+
+
+@dataclass
+class SwarmConfig:
+    """Configuration for swarm execution safety."""
+
+    max_handoffs: int = 10
+    max_iterations: int = 20
+    execution_timeout: float = 900.0  # Total execution timeout (seconds)
+    node_timeout: float = 300.0  # Individual node timeout (seconds)
+    ping_pong_check_nodes: int = 8  # Number of recent nodes to check for ping-pong
+    ping_pong_min_unique_nodes: int = 3  # Minimum unique nodes required in recent sequence
+
+
+@dataclass
+class SharedContext:
+    """Shared context accessible via tools."""
+
+    context: dict[str, dict[str, Any]] = field(default_factory=dict)
+    node_history: list[SwarmNode] = field(default_factory=list)
+    current_task: str | None = None
+    available_nodes: list[SwarmNode] = field(default_factory=list)
+
+    def set_task(self, task: str) -> None:
+        """Set the current task."""
+        self.current_task = task
+
+    def set_available_nodes(self, nodes: list[SwarmNode]) -> None:
+        """Set list of available agents."""
+        self.available_nodes = nodes
+
+    def add_context(self, node: SwarmNode, key: str, value: Any) -> None:
+        """Add context."""
+        if node.node_id not in self.context:
+            self.context[node.node_id] = {}
+        self.context[node.node_id][key] = value
+
+    def get_relevant_context(self, target_node: SwarmNode) -> dict[str, Any]:
```

Comment:
I'm presuming this will come from docs, but I'm not really understanding what "relevant context" is.